src/rust/engine/process_execution/bazel_protos/build.rs

```diff
@@ -14,6 +18,34 @@ fn main() {
     thirdpartyprotobuf.to_str().unwrap()
   );
 
+  let amended_proto_root = tempfile::TempDir::new().unwrap();
```

Comment:
Moving this new block out into a function with a docstring would be helpful to isolate/document its sideeffect.